### PR TITLE
mackup: init at 0.8.40

### PR DESCRIPTION
**Pull request by luftmensch-luftmensch**
## Description of changes
[Mackup](https://github.com/lra/mackup/) is a python application used to keep personal application settings in sync

Requested by GH-309951 

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ma/mackup/package.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  ];
+
+  propagatedBuildInputs = with python3Packages; [
+    setuptools
+    six
+    docopt
+  ];
+
+  # upstream has no tests
+  doCheck = false;
```
Comment (Misterio77) [inline pkgs/by-name/ma/mackup/package.nix:32]:
```suggestion
  checkPhase = "nosetests --with-coverage --cover-branches --cover-package=mackup";
```

File: pkgs/by-name/ma/mackup/package.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    rev = "refs/tags/${version}";
+    hash = "sha256-hAIl9nGFRaROlt764IZg4ejw+b1dpnYpiYq4CB9dJqQ=";
+  };
+
+  nativeBuildInputs = with python3Packages; [
+    poetry-core
+    pythonRelaxDepsHook
+    setuptools
+    docopt
+  ];
```
Comment (Misterio77) [inline pkgs/by-name/ma/mackup/package.nix:23]:
```suggestion
    nose
  ];
```

**Review by Misterio77 — CHANGES_REQUESTED** [review]
Hey, thanks for the PR!

Upstream does have tests, though. Here's how to run them.

And, actually, the failing test just pointed us at a issue we wouldn't have noticed otherwise, hardcoded `pgrep` binary: https://github.com/lra/mackup/blob/77e81127c718d12377889aa87b3a1c7501939143/mackup/utils.py#L286.

File: pkgs/by-name/ma/mackup/package.nix (hunk 1 above)
Comment (Misterio77) [reply pkgs/by-name/ma/mackup/package.nix:32]: Source: https://github.com/lra/mackup/blob/0.8.40/Makefile#L7

File: pkgs/by-name/ma/mackup/package.nix (hunk 2 above)
Comment (luftmensch-luftmensch) [reply pkgs/by-name/ma/mackup/package.nix:23]: Oh I've missed that. Thanks for pointing me out.

Adding those

**Comment by luftmensch-luftmensch** [comment]
@Misterio77 Updated! Should be resolved now

**Review by Misterio77 — APPROVED** [review]
LGTM!

File: pkgs/by-name/ma/mackup/package.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    rev = "refs/tags/${version}";
+    hash = "sha256-hAIl9nGFRaROlt764IZg4ejw+b1dpnYpiYq4CB9dJqQ=";
+  };
+
+  nativeBuildInputs = with python3Packages; [
+    poetry-core
+    pythonRelaxDepsHook
+    setuptools
+    docopt
+    nose
```
Comment (SuperSandro2000) [inline pkgs/by-name/ma/mackup/package.nix:24]: Shouldn't this go to nativeCheckInputs?

File: pkgs/by-name/ma/mackup/package.nix [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    docopt
+  ];
+
+  pythonImportsCheck = [ "mackup" ];
+
+  checkPhase = "nosetests --with-coverage --cover-branches --cover-package=mackup";
+  postPatch = ''
+    substituteInPlace mackup/utils.py \
+      --replace-fail '"/usr/bin/pgrep"' '"${lib.getExe' procps "pgrep"}"'
+  '';
```
Comment (SuperSandro2000) [inline pkgs/by-name/ma/mackup/package.nix:39]:
```suggestion
  checkPhase = ''
    nosetests
  '';
```
we don't care about coverage or even want to abort the build it is below XX

File: pkgs/by-name/ma/mackup/package.nix [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  version = "0.8.40";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "lra";
+    repo = "mackup";
+    rev = "refs/tags/${version}";
+    hash = "sha256-hAIl9nGFRaROlt764IZg4ejw+b1dpnYpiYq4CB9dJqQ=";
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/by-name/ma/mackup/package.nix:18]:
```suggestion

  postPatch = ''
    substituteInPlace mackup/utils.py \
      --replace-fail '"/usr/bin/pgrep"' '"${lib.getExe' procps "pgrep"}"'
  '';

```

File: pkgs/by-name/ma/mackup/package.nix [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  nativeBuildInputs = with python3Packages; [
+    poetry-core
+    pythonRelaxDepsHook
+    setuptools
```
Comment (SuperSandro2000) [inline pkgs/by-name/ma/mackup/package.nix:22]:
```suggestion
```
the package uses poetry-core as its build system

File: pkgs/by-name/ma/mackup/package.nix [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+  postPatch = ''
+    substituteInPlace mackup/utils.py \
+      --replace-fail '"/usr/bin/pgrep"' '"${lib.getExe' procps "pgrep"}"'
+  '';
+
+  nativeBuildInputs = with python3Packages; [
+    poetry-core
+    pythonRelaxDepsHook
+    docopt
```
Comment (SuperSandro2000) [inline pkgs/by-name/ma/mackup/package.nix:27]: I don't think we need a host docopt

```suggestion
```

Comment (luftmensch-luftmensch) [reply pkgs/by-name/ma/mackup/package.nix]: @SuperSandro2000 Applied suggestion!

**Pull request merged**